lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -1044,12 +1082,57 @@ overrideFeePolicy linearFee@(LinearFee a b _) override =
 poolRegistrationsFromBlock :: Block -> [W.PoolRegistrationCertificate]
 poolRegistrationsFromBlock (Block _hdr fragments) = do
     PoolRegistration (poolId, owners, taxes, _tx) <- fragments
-    let margin = fromRight maxBound $ mkPercentage @Int $ round $ 100 *
-               ( double (taxRatioNumerator taxes)
-               / double (taxRatioDenominator taxes)
-               )
+    let margin = fromRight maxBound $ mkPercentage @Int $ round $
+            100 * ratio (taxRatio taxes)
     let cost = Quantity (taxFixed taxes)
     pure $ W.PoolRegistrationCertificate poolId owners margin cost
+
+-- | If all incentives parameters are present in the blocks, returns a function
+-- that computes reward based on a given epoch.
+-- Returns 'Nothing' otherwise.
+rankingEpochConstants
+    :: Block
+    -> Maybe (EpochNo -> Quantity "lovelace" Word64 -> EpochConstants)
+rankingEpochConstants block0 = do
+    poolCapping   <- mPoolCapping
+    rewardLimit   <- mRewardLimit
+    treasuryTax   <- mTreasuryTax
+    rewardFormula <- mRewardFormula
+    pure $ \ep totalStake -> EpochConstants
+        { leaderStakeInfluence =
+            unsafeMkNonNegative 0
+        , desiredNumberOfPools =
+            unsafeMkPositive $ fromIntegral $ maxParticipation poolCapping
+        , totalRewards =
+            rewardsAt (rewardLimit, totalStake) treasuryTax ep rewardFormula
```

Comment:
super-nitpick: if the name is rewards**At**, the `epoch` should be the first argument, I think.